### PR TITLE
Add action to verify links on PR

### PR DESCRIPTION
**Pull request by rafvasq**
Created a workflow that runs the script to verify all document links upon a new PR

### CODE REVIEW

File: .github/workflows/verify-links.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    branches: 
+    - main
+    
+    tags:
+    - v*
+  
+  pull_request:
+
+env:
+  GITHUB_ACTION: "true"
```
Comment (ckadner) [inline .github/workflows/verify-links.yml:18]: Hi @rafvasq -- I don't think we need these `env` vars, do we?

File: .github/workflows/verify-links.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      
+      - name: Install dependencies
+        run: pip install requests
+
+      - name: Link check
+        run: |
+          ./tools/python/verify_doc_links.py
```
Comment (ckadner) [inline .github/workflows/verify-links.yml:43]: missing a new line at the end of file

**Review by ckadner — CHANGES_REQUESTED** [review]
Hi @rafvasq -- thank you for your PR and sorry for not reviewing it sooner. Looks good. Just 2 questions/comments

File: .github/workflows/verify-links.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (ckadner) [inline .github/workflows/verify-links.yml:38]: weird, it still says `No newline at end of file`

Comment (rafvasq) [reply .github/workflows/verify-links.yml]: definitely something strange on my side, but looks resolved now 😅

**Review by ckadner — APPROVED** [review]
/lgtm

**Comment by ckadner** [comment]
Looks good! Thank you @rafvasq

**Pull request merged**